lib/user_session_management.rb

```diff
@@ -35,7 +35,14 @@ def current_user
     @current_user ||= begin
       user_hash = sso_cookie_jar.subject || old_sso_cookie_jar[old_sso_cookie_name]&.[]('user')
       current_sso_user = User.find_by(uuid: user_hash['uuid']) if user_hash.present?
-      current_sso_user.nil? ? AnonymousUser.instance : current_sso_user
+
+      if current_sso_user.nil?
+        AnonymousUser.instance
+      else
+        # Some users may not have the new SSO cookie yet, so set it
+        sso_cookie_jar.subject = user_hash if sso_cookie_jar.subject.blank?
```

Comment:
This is the only new line.  The other “new” lines are just restating the `... ? ... : ...` ternary operator as an `if else`